src/types.rs

```diff
@@ -340,37 +341,59 @@ impl RpcAccess {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, CandidType, Serialize, Deserialize, Default)]
-pub enum LogMessageFilter {
+pub enum ConsoleFilter {
     #[default]
     ShowAll,
-    ShowOnly(Vec<LogMessageType>),
+    HideAll,
+    ShowPattern(RegexString),
+    HidePattern(RegexString),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, CandidType, Serialize, Deserialize, Default)]
+pub struct RegexString(String);
+
+impl RegexString {
+    pub fn is_match(&self, value: &str) -> bool {
+        Regex::new(&self.0)
```

Comment:
Note that this will compile the regex every time a message is logged, which is a somewhat expensive operation. If we wanted to optimize thing, we should store the compiled Regex in stable memory, but the underlying structure appears complex and that might not be so easy. Given that this is just for local debugging I don't think this is worth it and would leave as is.

Reply:
That was my reasoning as well. Added a comment explaining why the regex is currently compiled every time it's used.